platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/style/ZoomFunctionSymbolLayerActivity.java

```diff
@@ -119,7 +128,8 @@ public void onMapClick(@NonNull LatLng point) {
         if (!featureList.isEmpty()) {
           Feature feature = featureList.get(0);
           boolean isSelected = feature.getBooleanProperty(KEY_PROPERTY_SELECTED);
-          updateSource(!isSelected);
+          ZoomFunctionSymbolLayerActivity.this.isSelected = !isSelected;
```

Comment:
What about changing the name of the local `isSelected` variable so you don't need to be explicit (`ZoomFunctionSymbolLayerActivity.this.isSelected`)? Maybe `isFeatureSelected`?